### PR TITLE
Add a Guide for Sequencial and Parallel Calls

### LINKED ISSUE #89: Write a guide about doing sequence and parallel calls
Write a guide about how one can do sequence and parallel calls in Tailcall. Tailcall automatically identifies the fastest way to get data. If something can be done in parallel, it will happen automatically without the user explicitly marking something as parallel. Write a guide about how this happens.

## Content Creation Requirements
To maintain the quality of our content, please adhere to the following guidelines:

### 1. Accuracy in Language
- **Grammar and Spelling:** Ensure your content is free from grammatical and spelling errors.

### 2. Tone and Style
- **Neutral Tone:** Maintain a neutral, non-emotional tone throughout the content.
- **Engaging Style:** Write in a free-flowing and engaging manner, keeping the reader's interest.

### 3. Content Integrity
- **Fact-Checking:** Verify all information. If unsure, use Discord to clarify.
- **Relevance:** Ensure all content is cohesive, to the point, and directly related to the title.

### 4. Originality
- **Avoid Low-Effort Content:** Content should be original and not solely generated by AI tools like ChatGPT.

**PS: Adherence to these guidelines is crucial. Content not meeting these standards may require revision or may not be accepted.**

### PR DESCRIPTION
**Pull request by tareknaser**
Added `docs/guides/calls.md` for sequential and parallel calls management in Tailcall documentation.

/closes #89
/claim #89

### CODE REVIEW

File: docs/guides/calls.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
+---
+title: "Seamless Sequential and Parallel Calls in Tailcall"
```
Comment (tusharmath) [inline docs/guides/calls.md]:
```suggestion
title: "Sequencing & Parallelism"
description: "How Tailcall seamlessly sequences and parallelizes API calls for your applications, enhancing efficiency without any manual input"
```

File: docs/guides/calls.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+}
+
+type Post {
+  id: Int!
+  userId: Int!
+  title: String!
+  body: String!
+  user: User
+    @http(path: "/users", query: [{key: "id", value: "{{value.userId}}"}], matchPath: ["id"], matchKey: "userId")
+}
```
Comment (tusharmath) [inline docs/guides/calls.md]: We don't have `matchPath` and `matchKey` anymore. I think in this example we should talk about the n + 1 usecase. Where for each post we are querying it's user. We should emphasize on the fact that tailcall automatically identifies that all the requests to `/users/{{value.userId}}` should go in parallel.

File: docs/guides/calls.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    user {
+      id
+      name
+    }
+  }
+}
+```
+
+- This query retrieves a post with its comments and their corresponding authors.
+- Implicit Sequencing: Tailcall understands that retrieving a comment's author details depends on first knowing the comment's ID. Hence, it automatically fetches the comments first, then uses their IDs to fetch the respective users in a sequential manner.
```
Comment (tusharmath) [inline docs/guides/calls.md]: Can format this better, without bullet points.

File: docs/guides/calls.md (hunk 2 above)
Comment (tusharmath) [reply docs/guides/calls.md]: This could be an example of making parallel calls.

File: docs/guides/calls.md [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+### Example 2: Searching Multiple Posts and Users by ID
```
Comment (tusharmath) [inline docs/guides/calls.md]: Add another example of making sequence calls where we query one user and then all it's posts.

File: docs/guides/calls.md [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+  users(ids: $userIds) {
+    id
+    name
+    posts(ids: $userIds) {
+      id
+      title
+      body
+    }
+  }
+}
```
Comment (tusharmath) [inline docs/guides/calls.md]: This query isn't possible. We don't have `posts` on the type `User` in the above example.

File: docs/guides/calls.md (hunk 2 above)
Comment (tareknaser) [reply docs/guides/calls.md]: Sounds good.
I moved this example to be the third to keep the first 2 examples more straightforward

File: docs/guides/calls.md (hunk 4 above)
Comment (tareknaser) [reply docs/guides/calls.md]: Added as the first example. 
Let me know if that's what you had in mind

File: docs/guides/calls.md (hunk 5 above)
Comment (tareknaser) [reply docs/guides/calls.md]: Yes! I think this one slipped 
Thanks for the catch

**Review by tusharmath — APPROVED** [review]

**Comment by tusharmath** [comment]
Very clear and nicely explained! Thanks!

**Pull request merged**